core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala

```diff
@@ -311,25 +311,53 @@ abstract class ApiDocDirective(name: String)
 }
 
 object ApiDocDirective {
-  /** This relies on the naming convention of packages being all-ascii-lowercase (which is rarely broken), numbers and underscore. */
-  def packageDotsToSlash(s: String) = s.replaceAll("(\\b[a-z][a-z0-9_]*)\\.", "$1/")
+  /**
+   * Converts package dot notation to a path, separated by '/'
+   * Allow all valid java characters and java numbers to be used, according to the java lang spec. In order to
+   * be allow to convert `package.Outer.Inner` in scala to the `packge/Outer$$Inner`, `strictPackageIdent` has
+   * be set to `true`.
```

Comment:
Seems like this is no longer accurate. Good to give an example: `startWithLowercase` will get it wrong when a package name starts with an uppercase letter or when an inner class starts with a lowercase character, while `startWithAnycase` will derive the wrong path whenever an inner class is encountered. Does that summarize the trade-off correctly?

Reply:
You're right, it's the old style of setting. Will shorten this block and add your description to the `@param` description.

```suggestion
   * Converts package dot notation to a path, separated by '/'
   * Allow all valid java characters and java numbers to be used, according to the java lang spec.
   *
   * @param s package or full qualified class name to be converted.
   * @param packageNameStyle Setting `startWithLowercase`` will get it wrong when a package name
   *                         starts with an uppercase letter or when an inner class starts with
   *                         a lowercase character, while `startWithAnycase` will derive the wrong
   *                         path whenever an inner class is encountered.
```